openbas-api/src/main/java/io/openbas/xtmhub/collector/XtmHubCollector.java

```diff
@@ -0,0 +1,28 @@
+package io.openbas.xtmhub.collector;
+
+import io.openbas.xtmhub.XtmHubService;
+import jakarta.annotation.PostConstruct;
+import java.time.Duration;
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class XtmHubCollector {
+
+  private final XtmHubCollectorConfig config;
+  private final ThreadPoolTaskScheduler taskScheduler;
+  private final XtmHubService xtmHubService;
+
+  @PostConstruct
+  public void init() {
```

Comment:
After reviewing and discussions with staffs, there might be a need to rework this scheduler here. Will keep you updated asap.